blockchain/v2/routine.go

```diff
@@ -0,0 +1,194 @@
+package v2
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// TODO
+// * revisit panic conditions
+// * audit log levels
+// * Convert routine to an interface with concrete implmentation
+
+type handleFunc = func(event Event) (Events, error)
+
+type Routine struct {
```

Comment:
Could you explain the rational for this struct?

Reply:
The intention for routines is to provide lifecycle management (start/stop) behaviour to 
 [adr-30](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-030-consensus-refactor.md) style finite state machines characterized by functions of the form:
```go
type handleFunc = func(event Event) (Events, error)
```